### ISSUE TITLE
Doesn't support reactions with custom emoji from other servers

### ISSUE THREAD

**Issue by theflyinggrayson** [BtoBastian/Javacord#101]
As seen here, reactions automatically assume that the custom emoji are from the server the channel is sent from, meaning that reactions with custom emoji that aren't from that server aren't added to the Message entity.
```
public ImplReaction(ImplDiscordAPI api, Message message, JSONObject data) {
      -snip-
        if(emoji.isNull("id")) {
            this.customEmoji = null;
            this.unicodeEmoji = emoji.getString("name");
        } else {
            this.unicodeEmoji = null;
            this.customEmoji = message.getChannelReceiver().getServer().getCustomEmojiById(emoji.getString("id"));
        }
}

**Comment by BtoBastian** [comment]
Hey,
this is already considered/fixed in the current rewrite to version 3.0.0.
I won't change it in v2, as v3 will replace v2 soon™ anyways.
If you want some further information about the rewrite, fell free to join the Javacord server https://discord.gg/0qJ2jjyneLEgG7y3 :)

**Comment by theflyinggrayson** [comment]
Oh, awesome. I didn't even see that branch, thanks for the info
